streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -206,14 +202,14 @@ public static WindowBytesStoreSupplier persistentWindowStore(final String name,
                                                                  final boolean retainDuplicates,
                                                                  final long segmentInterval) {
         Objects.requireNonNull(name, "name cannot be null");
-        if (retentionPeriod < 0) {
+        if (retentionPeriod < 0L) {
             throw new IllegalArgumentException("retentionPeriod cannot be negative");
         }
-        if (windowSize < 0) {
+        if (windowSize < 0L) {
             throw new IllegalArgumentException("windowSize cannot be negative");
         }
-        if (segmentInterval < 60_000) {
-            throw new IllegalArgumentException("segmentInterval must be at least one minute");
+        if (segmentInterval < 0L) {
```

Comment:
I set the min to 0, following the other two bounds, but wouldn't 0 cause problems in practice? Should we instead set all three minima to 1?

Reply:
I'd suggest setting the min to 0 other than 1, because interval maybe used as denominator in some use cases?